### ISSUE TITLE
Rule Enhancement Proposal: curly allow single line that throws or return

### ISSUE THREAD

**Issue by vtfn** [eslint/eslint#6552]
I think it's quite common developers write conditions like
```javascript
if (shouldStopHere) return
```
or

``` javascript
if (oops) throw new Error()
```

The problem is that sometimes either the condition or the statment is too long and keeping it in a single line affects legibility, so we end up spliting it in two lines

``` javascript
if (shouldStopHere || weGotAnError || whateverReason)
  throw new SomeKindOfStrangeError('Something very weird just happened')
```

Or, and that is only a personal preference, return something other undefined would be better done in a new line

``` javascript
if (shouldStopHereWithThisValue)
  return foo(true)
// keep going
return foo(false)
```

In all these cases we are sure that nothing will ever be added into the body of that if... so there is no need for braces and omitting them save us some lines.

That said I think it would be good to have an option that disable ```curly``` for a single return or throw statement.

**Comment by alberto** [comment]
Thanks for your interest in improving eslint. Unfortunately, it looks like consensus couldn't be reached on this issue and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after 21 days tend never to reach consensus, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
